ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/WsAgentStateController.java

```diff
@@ -151,63 +188,92 @@ public void makeCall(AsyncCallback<MessageBus> callback) {
-        asyncRequestFactory.createGetRequest(url)
-                           .send(new StringUnmarshaller())
-                           .then(new Operation<String>() {
-                               @Override
-                               public void apply(String result) throws OperationException {
-                                   JSONObject object = null;
-                                   try {
-                                       object = JSONParser.parseStrict(result).isObject();
-                                   } catch (Exception exception) {
-                                       Log.warn(getClass(), "Parse root resources failed.");
-                                   }
-
-                                   if (object != null && object.containsKey("rootResources")) {
-                                       JSONArray rootResources = object.get("rootResources").isArray();
-                                       for (int i = 0; i < rootResources.size(); i++) {
-                                           JSONObject rootResource = rootResources.get(i).isObject();
-                                           String regex = rootResource.get("regex").isString().stringValue();
-                                           String fqn = rootResource.get("fqn").isString().stringValue();
-                                           String path = rootResource.get("path").isString().stringValue();
-                                           availableServices.add(new RestServiceInfo(fqn, regex, path));
-                                       }
-                                   }
-
-                                   checkWsConnection();
-                               }
-                           })
-                           .catchError(new Operation<PromiseError>() {
-                               @Override
-                               public void apply(PromiseError arg) throws OperationException {
-                                   Log.error(getClass(), arg.getMessage());
-                                   new Timer() {
+        //here we add trailing slash because {@link org.eclipse.che.api.core.rest.ApiInfoService} mapped in this way
+        String url = devMachine.getWsAgentBaseUrl() + '/';
+        final String pingUrl = restContext + "/workspace-agent-health/" + devMachine.getWorkspace() + '/' + WSAGENT_REFERENCE;
+        asyncRequestFactory.createGetRequest(url).send(new StringUnmarshaller()).then(new Operation<String>() {
+            @Override
+            public void apply(String result) throws OperationException {
+                JSONObject object = null;
+                try {
+                    object = JSONParser.parseStrict(result).isObject();
+                } catch (Exception exception) {
+                    Log.warn(getClass(), "Parse root resources failed.");
+                }
+
+                if (object != null && object.containsKey("rootResources")) {
```

Comment:
if you move this if-block to the try-block upper
checking _object != null_ will be unnecessary